### PR TITLE
Json edit

### PR DESCRIPTION
**Pull request by TheGuyWhoCodes**
## Majors
 - Add a "comment section" with comments coming from Datastore. The format it comes back as is ArrayList<HashMap<String, String>>.
 - Form grabs comments on every submit. This means it'll automatically refresh to grab new comments
- Uses JavaScript ajax instead of PHP POST
- Rate limiting requests if many requests come in at once.
## Minors
 - Adds Google Guava to Maven, used for Rate Limiting

## Questions
 - I'm not a fan of the way I parse the Entity into a JSON readable format. I was looking to use the Java stream API in order to have constant time conversion, but was unable to figure out how to get that working with the Entity format. Does anyone have some feedback on a way to approach this? Is there maybe a way to map an Entity into a POJO?

## Screenshot
https://screenshot.googleplex.com/u0AkHRsEg02.png

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -19,14 +19,36 @@
...
+    	response.setContentType("application/json;");
+        Gson gson = new Gson();
+        response.getWriter().println(gson.toJson(comments));
+    }
+
+	@Override
+  	public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        response.setContentType("application/json;");
+        boolean isAcquired = rateLimiter.tryAcquire();
+        if(!isAcquired) {
```
Comment (ghost) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Please explain with a comment why you need a rate limiter and what will happen if it is not acquired

Comment (TheGuyWhoCodes) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Kiran, with PR 3894a90, I added Java comments to the doGet and doPost. 

The Rate Limiter will pause that request if it goes over the limit until the rate limiter resets.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -18,15 +18,89 @@
...
+    final RateLimiter commentPostLimiter = RateLimiter.create(10.0); 
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
 
-  @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("Hello from Chris!!");
-  }
+	/**
+    *   this doGet() function will write out the List of comments to JSON
```
Comment (ghost) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:52]: Let us talk about the Ratelimiter offline.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
Comment (TheGuyWhoCodes) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Per this document: https://cloud.google.com/datastore/docs/best-practices

and https://cloud.google.com/datastore/docs/concepts/limits

It looks like removing the rate limiter will be the best option.

**Review by ghost — APPROVED** [review]

**Pull request merged**